package/gargoyle/files/www/js/access.js

```diff
@@ -12,26 +12,22 @@ var stopRedirect = false;
 
 function saveChanges()
 {
-	errorList = proofreadAll();
+	var errorList = proofreadAll();
```

Comment:
You have added "var" in several places where it is not explicitly necessary. Are you doing this intentionally or do you have an IDE doing things automatically?
These take up 4 bytes each in flash which is at a premium. Consider removal if not explicitly needed.

Reply:
Interesting, haven't thought of saving space here. It's just vim and was intentionally. I will remove it where possible.

Reply:
Just nit picking here but if those 4 bytes are a problem then the js files would be better off being properly minified; common.js script for example could save ~30% on the file size and access.js around 25%.

Reply:
> Just nit picking here but if those 4 bytes are a problem then the js files would be better off being properly minified; common.js script for example could save ~30% on the file size and access.js around 25%.

They do get minified during compilation. The 4 bytes do not get factored out as the minifier does not have enough context to factor them out.

Reply:
I played a bit with uglifyjs options. I think @DoubleOhmSeven meant we can do better than now by enabling `-c -m` flags.


|file              | now  |  -c -m |   %   |
|-------------  | -------  |-------  |------|
|common.js |60 KiB | 42 KiB | 30  |
|access.js    |24 KiB | 18 KiB | 25  |

By doing so, I also observed that global variable names are not minified without the `--toplevel` option, making the file size even larger when removing `var`s. I want to try the `--name-cache` option as well.

Reply:
Correct, but the minifier used is pretty terrible, if you use UglifyJS 3 (you can check https://jscompress.com/ for difference) then you'll notice a huge difference. For example, common.js of 1.12.0: 31.89% compression, saving 18.54 kb.

Reply:
I've also just found [Terser](https://github.com/terser/terser):

> `uglify-es` is [no longer maintained](https://github.com/mishoo/UglifyJS2/issues/3156#issuecomment-392943058) and `uglify-js` does not support ES6+.

> **`terser`** is a fork of `uglify-es` that mostly retains API and CLI compatibility
with `uglify-es` and `uglify-js@3`.

Reply:
If there is a better or more up to date uglifyjs/CSS that will give additional savings, im happy to accept a patch or suggestions. 
But in my brief reading, I would be quite sceptical of it operating in a non breaking way with function mangling and some global's (i18n vars) given the way the js is not internally referenced (i.e. the files are included by outputs of the header footer program). So it would need to be thoroughly tested and shown working.

Reply:
Okay, I will give it a try and check what impact mangling will have.